query/src/sql/planner/binder/select.rs

```diff
@@ -260,7 +271,10 @@ impl<'a> Binder {
                 Ok((s_expr, bind_context))
             }
             TableReference::Join(join) => self.bind_join(bind_context, join).await,
-            _ => Err(ErrorCode::UnImplement("Unsupported table reference type")),
+            TableReference::Subquery { subquery, alias } => {
+                self.bind_query_with_alias(subquery, alias, bind_context)
+                    .await
+            }
```

Comment:
I think you can just call `apply_table_alias` here.

Reply:
Do you mean directly processing columns of current `bind_context`?
```rust
        if let Some(table_alias) = table_alias {
            for column_binding in bind_context.clone().all_column_bindings().iter() {
                if let Some(origin_table_name) = &column_binding.table_name {
                    bind_context.apply_table_alias(origin_table_name, table_alias)?;
                }
            }
        }
```

Reply:
Yes, this work has been done by `apply_table_alias`. Actually, the argument `original_name` is useless now, you can remove it safely.